AnnService/inc/Core/Common/Labelset.h

```diff
@@ -13,31 +13,62 @@ namespace SPTAG
     {
         class Labelset
         {
+        public:
+            enum class InvalidIDBehavior
+            {
+                Passthrough,
+                AlwaysContains,
+                AlwaysNotContains
+            };
         private:
             std::atomic<SizeType> m_inserted;
             Dataset<std::int8_t> m_data;
+            InvalidIDBehavior m_invalidIDBehaviorSetting;
             
         public:
-            Labelset() 
+            Labelset()
             {
                 m_inserted = 0;
                 m_data.SetName("DeleteID");
             }
 
-            void Initialize(SizeType size, SizeType blockSize, SizeType capacity)
+            void Initialize(SizeType size, SizeType blockSize, SizeType capacity, InvalidIDBehavior invalidIDBehaviorSetting = InvalidIDBehavior::Passthrough)
             {
+                m_invalidIDBehaviorSetting = invalidIDBehaviorSetting;
                 m_data.Initialize(size, 1, blockSize, capacity);
             }
 
             inline size_t Count() const { return m_inserted.load(); }
 
             inline bool Contains(const SizeType& key) const
             {
+                if (key >= R())
+                {
+                    switch (m_invalidIDBehaviorSetting)
+                    {
+                        case InvalidIDBehavior::AlwaysContains:
+                            return true;
+                        case InvalidIDBehavior::AlwaysNotContains:
+                            return false;
+                    }
+                }
+
                 return *m_data[key] == 1;
             }
 
             inline bool Insert(const SizeType& key)
             {
+                if (key >= R())
+                {
+                    switch (m_invalidIDBehaviorSetting)
+                    {
+                        case InvalidIDBehavior::AlwaysContains:
+                            return true;
+                        case InvalidIDBehavior::AlwaysNotContains:
+                            return false;
```

Comment:
need default

Reply:
the default case is Passthrough, it falls through to the old logic (dataset class throwing exception)